clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1752,8 +1754,10 @@ static void AppendTargetClonesMangling(const CodeGenModule &CGM,
   const TargetInfo &TI = CGM.getTarget();
   if (TI.getTriple().isAArch64()) {
```

Comment:
Can you please explain why AArch64 needs to be 'specially' handled here? I can't recall where but other targets get a .default mangling as well right? Why isn't AArch64 being handled wherever it is happening for other targets?

Reply:
As I see for other targets `"default"` is just copied as a feature string.
For AArch64 we want to have clones per feature or set of features instead of CPU.
```c
__attribute__((target_clones("crc32", "aes+sha1")))
int foo(){..}
```

would give 3 clones; the default, `crc32` and one for `aes` plus `sha1` mangled as `foo.default`, `foo._Mcrc32`,` foo._Msha1Maes` respectively.
Also "aes+sha1" should be the same as "sha1+aes" after mangling.

Reply:
@DanielKristofKiss slightly off topic: it's too bad `default` is implied, as it makes `target_clones` incompatible with `target_version` for constructs like this:

```
__attribute__((target_clones("dotprod", "aes"))) 
int callee(void) {
    return 42;
}

__attribute__((target_version("default"))) 
int callee(void) {
    return 0;
}

int caller(void) {
    return callee();
}
```

Reply:
@jroelofs currently in [FMV spec](https://github.com/ARM-software/acle/blob/main/main/acle.md#function-multi-versioning):
```
The following attributes trigger the multi version code generation: __attribute__((target_version("name"))) and __attribute__((target_clones("name",...))).
These attributes can't be mixed with each other.
```

Reply:
Right, it's too bad this incompatibility is baked into the spec. Are there other reasons why this can't be supported?

Reply:
It was not supported in the original function multi versioning neither and excluded to that as we building top of it.
Anyway created a ticket for it https://github.com/ARM-software/acle/issues/291 . I need to check what GCC and others say about it.